### PR TITLE
Keras file format updates

### LINKED ISSUE #88: Resolve Keras File Format Issues
TensorFlow updated to version 2.16.1 on March 8, 2024. Python 3.9+ tests in the main idaes-pse repository that install the latest package and run examples now throw errors similar to the message below:

```
E           ValueError: File format not supported: filepath=D:\a\idaes-pse\idaes-pse\idaes\core\surrogate\plotting\tests\keras_surrogate. Keras 3 only supports V3 `.keras` files and legacy H5 format files (`.h5` extension). Note that the legacy SavedModel format is not supported by `load_model()` in Keras 3. In order to reload a TensorFlow SavedModel as an inference-only layer in Keras 3, use `keras.layers.TFSMLayer(D:\a\idaes-pse\idaes-pse\idaes\core\surrogate\plotting\tests\keras_surrogate, call_endpoint='serving_default')` (note that your `call_endpoint` might have a different name).
```

Examples that use Keras should be updated accordingly.

### PR DESCRIPTION
**Pull request by bpaul4**
Fixes #88.

Dependent on https://github.com/IDAES/idaes-pse/pull/1401 for tests to pass.

----
Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

    I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
    I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

----
📚 Documentation preview 📚: https://idaes-examples--128.org.readthedocs.build/en/128/

### CODE REVIEW

File: idaes_examples/notebooks/docs/surrogates/sco2/omlt/surrogate_embedding_usr.ipynb
Comment (lbianchi-lbl) [inline idaes_examples/notebooks/docs/surrogates/sco2/omlt/surrogate_embedding_usr.ipynb:446]: @bpaul4 was this change required because of a specific error? Usually I'd expect (but haven't actually checked) that Unix-style paths with `/` are accepted on Windows as well, so I'd be curious to know if there might be something that warranted this change.

Comment (bpaul4) [reply idaes_examples/notebooks/docs/surrogates/sco2/omlt/surrogate_embedding_usr.ipynb:446]: That's very odd. The "\\" change doesn't exist on my local branch; there's another change in "_doc" version of this notebook from ".ipynb" to ".md" that I am sure I did not make. Not sure how this happened, as the other versions should all be identical to the source notebook, correct?

File: requirements-dev.txt
Comment (lbianchi-lbl) [inline requirements-dev.txt:5]: FIXME: revert this before merging

**Review by lbianchi-lbl — CHANGES_REQUESTED** [review]
Just a reminder to revert the changes to `requirements-dev.txt` before merging.

File: idaes_examples/notebooks/docs/surrogates/sco2/omlt/surrogate_embedding_usr.ipynb
Comment (lbianchi-lbl) [reply idaes_examples/notebooks/docs/surrogates/sco2/omlt/surrogate_embedding_usr.ipynb:446]: I agree, that's very strange. I'll try to look into it, starting from attempting to reproduce this locally.

**Comment by lbianchi-lbl** [comment]
We should proceed as follows:

1. Wait until IDAES/idaes-pse#1401 is merged
2. Revert the change to the requirements file causing IDAES to be installed from the PR branch
3. Merge this in (assuming tests pass)

**Comment by bpaul4** [comment]
The 3.8 tests appear stuck, and everything else is passing. @lbianchi-lbl is the plan to remove 3.8 support in this repository, and remove the 3.8 tests?

**Comment by lbianchi-lbl** [comment]
> The 3.8 tests appear stuck, and everything else is passing. @lbianchi-lbl is the plan to remove 3.8 support in this repository, and remove the 3.8 tests?
> 
> FYI per @andrewlee94 we probably need to merge this PR first, and then 1401 in idaes-pse.

Good point @bpaul4, yes, let me open a PR for that.

**Pull request merged**
